### ISSUE TITLE
automatic logout

### ISSUE THREAD

**Issue by benzaked** [IQSS/dataverse#6404]
The system forwards to the new page unlogin after clicking on some button. for example:
login > add data.throw the user to the login page.
Or login> click on the logo > getting to the root dataverse unlogin
Does anyone know why?
 I install dataverse via docker from [here](https://github.com/IQSS/dataverse-docker) and also opened an issue [there](https://github.com/IQSS/dataverse-docker/issues/31) but got no answer.
Thanks!

**Comment by pdurbin** [comment]
@benzaked hi, in the last issue you opened, I didn't give you a very clear reply: https://github.com/IQSS/dataverse/issues/6384#issuecomment-555976065

Let me try again. 😄 

It is critically important to patch Weld as described at http://guides.dataverse.org/en/4.18.1/installation/prerequisites.html#installing-glassfish or else your installation may suffer from automatic logouts and even users sessions being scrambled (!) as described in #647.

dataverse-docker is a community effort. I see that the lead of that project, @4tikhonov, just replied to you at https://github.com/IQSS/dataverse-docker/issues/31#issuecomment-558173466 (thanks, Slava!)

Can you install Dataverse on a public URL and give us a link so that we can do our own testing? If we can confirm the bug on that server, would you be able to give us ssh access so we can poke around and check on Weld and other configuration?

**Comment by pdurbin** [comment]
@benzaked all that Weld stuff I mentioned was for Glassfish 4. Now that we're on Payara 5 we don't have to patch Weld at all. Are you still having problems? What version of Dataverse are you using these days?
